src/World.cpp

```diff
@@ -542,6 +542,30 @@ void cWorld::ChangeWeather(void)
 
 
 
+bool cWorld::IsWeatherWetAtBlock(Vector3i a_Pos)
+{
+	ASSERT(cChunkDef::IsValidHeight(a_Pos.y));
```

Comment:
Logically, this function makes no sense being called when outside of the world, as there are no blocks at which the weather could be wet. However, it would be possible to make this return the result of IsWeatherWetAt() if the Y coord is outside of the world. I don't know if that's really sensible or not, do you have an opinion?

Reply:
Maybe the name maybe changed from `Block` to `Pos` or is that too vague?

As for the behaviour, in my opinion:
 * anything above the first rain blocker should be wet, including above the world.
 * anything below the first rain blocker should not be wet, including below the world.

One thing I'm not sure about is: if there are no rain blockers in a column, should positions under the world be considered wet?

Reply:
Both `Pos` and `Block` seem a bit vague. Maybe `IsWeatherWetAtXYZ`?

Your behaviour seems good, and I think the most sensible option is to consider positions under the world wet if there are no rain blockers above.

Reply:
`IsWeatherWetAtXYZ` sounds all right.  Maybe a name that tells the reader it considers rain blocking would be better but I can't think of anything now.

I tried it out in game and endermen won't teleport if it starts raining while they're below the world which suggests they're not considered wet.  This also matches up with the particle effects.  Are there any other interactions with rain I could check this with?

Reply:
The only other places IsWeatherWetAtBlock is used currently is for burning entities. 

If the behaviour there is consistent I'll change the behaviour below the world.

Reply:
Alright, I've had a chance to check and entities do continue to burn below the world.

Reply:
OK, great. I'll change the code, and the name.